src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -1,186 +1,182 @@
+import { DOMAINS_HIDE_MORE_INFO } from "../../../common/const.js";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import createRowElement from "../common/create-row-element.js";
+import computeDomain from "../../../common/entity/compute_domain.js";
+import processConfigEntities from "../common/process-config-entities";
+import { HomeAssistant } from "../../../types.js";
+import { EntityConfig, EntityRow } from "../entity-rows/types.js";
+
+interface ConfigEntity extends EntityConfig {
+  type?: string;
+  secondary_info: "entity-id" | "last-changed";
+  action_name?: string;
+  service?: string;
+  service_data?: object;
+  url?: string;
+}
+
+interface Config extends LovelaceConfig {
+  show_header_toggle?: boolean;
+  title?: string;
+  entities: ConfigEntity[];
+}
+
+class HuiEntitiesCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected _hass?: HomeAssistant;
+  protected _config?: Config;
+  protected _configEntities?: ConfigEntity[];
+
+  set hass(hass) {
+    this._hass = hass;
+    this.shadowRoot!.querySelectorAll("#states > *").forEach(
+      (element: unknown) => {
+        (element as EntityRow).hass = hass;
+      }
+    );
+  }
+
+  static get properties(): PropertyDeclarations {
+    return {
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    if (!this._config) {
+      return 0;
+    }
+    // +1 for the header
+    return (this._config.title ? 1 : 0) + this._config.entities.length;
+  }
+
+  public setConfig(config: Config) {
+    const entities = processConfigEntities(config.entities);
+    for (const entity of entities) {
+      if (
+        entity.type === "call-service" &&
+        (!entity.service || !entity.name || !entity.icon || !entity.action_name)
+      ) {
+        throw new Error("Missing required property when type is call-service");
+      } else if (
+        entity.type === "weblink" &&
+        (!entity.name || !entity.icon || !entity.url)
+      ) {
+        throw new Error("Missing required property when type is weblink");
+      }
+    }
+
+    this._config = config;
+    this._configEntities = entities;
+  }
+
+  protected render() {
+    if (!this._config || !this._hass) {
+      return html``;
+    }
+    const { show_header_toggle, title } = this._config;
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card>
+        ${
+          !title && !show_header_toggle
+            ? html``
+            : html`
+            <div class='header'>
+              <div class="name">${title}</div>
+              ${
+                show_header_toggle === false
+                  ? html``
+                  : html`
+                  <hui-entities-toggle
+                    .hass="${this._hass}"
+                    .entities="${this._configEntities!.map(
+                      (conf) => conf.entity
+                    )}"

```

Comment:
add `.filter(Boolean)` to filter out any `undefined` which are returned when fetching `entity` from weblink or service call config.